### PR TITLE
[ioctl/newcmd] add sm2 for account create and delete

### PR DESCRIPTION
**Pull request by huof6829**
modify as ioctl/new/account

### CODE REVIEW

File: ioctl/newcmd/account/accountdelete.go [hunk 1]
```diff
@@ -81,7 +79,7 @@ func NewAccountDelete(c ioctl.Client) *cobra.Command {
 	resultSuccess, _ := c.SelectTranslation(resultSuccess)
 	failToFindAccount, _ := c.SelectTranslation(failToFindAccount)
 
-	ad := &cobra.Command{
+	cmd := &cobra.Command{
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountdelete.go:82]: `return &cobra.Command{`

File: ioctl/newcmd/account/accountdelete.go [hunk 2]
```diff
@@ -99,43 +97,53 @@ func NewAccountDelete(c ioctl.Client) *cobra.Command {
...
-					fmt.Scanf("%s", &confirm)
-					if !strings.EqualFold(confirm, "yes") {
-						output.PrintResult("quit")
-						return nil
-					}
 
-					if err := os.Remove(v.URL.Path); err != nil {
-						return output.NewError(output.WriteFileError, failToRemoveKeystoreFile, err)
+			var filePath string
+			if CryptoSm2 {
```
Comment (CoderZhi) [inline ioctl/newcmd/account/accountdelete.go:102]: Move CryptoSm2 into ioctl.Client

File: ioctl/newcmd/account/accountdelete.go (hunk 1 above)
Comment (huof6829) [reply ioctl/newcmd/account/accountdelete.go:82]: yes

File: ioctl/newcmd/account/accountdelete.go (hunk 2 above)
Comment (huof6829) [reply ioctl/newcmd/account/accountdelete.go:102]: CryptoSm2 is just set in package account.  if move into ioctl.Client, this mean other package can set it, this is not suitable, i think

File: ioctl/newcmd/account/accountdelete.go [hunk 3]
```diff
@@ -99,43 +97,53 @@ func NewAccountDelete(c ioctl.Client) *cobra.Command {
...
-						aliases[addr] = name
-					}
+			// check whether crypto file exists
+			if _, err = os.Stat(filePath); err != nil {
+				return output.NewError(output.ReadFileError, fmt.Sprintf(failToFindAccount, addr), nil)
+			}
+			var confirm string
+			message := output.ConfirmationMessage{Info: infoWarn, Options: []string{"yes"}}
+			fmt.Println(message.String())
+			fmt.Scanf("%s", &confirm)
```
Comment (Liuhaai) [inline ioctl/newcmd/account/accountdelete.go:123]: use `AskToConfirm()`

File: ioctl/newcmd/account/accountdelete.go (hunk 2 above)
Comment (Liuhaai) [reply ioctl/newcmd/account/accountdelete.go:102]: `CryptoSm2` is unnecessary to be a global var

Comment (huof6829) [reply ioctl/newcmd/account/accountdelete.go:102]: it is called by  ioctl/newcmd/action  and  xctl.go  (both are not added now)

Comment (Liuhaai) [reply ioctl/newcmd/account/accountdelete.go:102]: The code in newcmd can used the new `CryptoSm2` from client. And keep the old one until it isn't depended

Comment (huof6829) [reply ioctl/newcmd/account/accountdelete.go]: Move CryptoSm2 into ioctl.Client

File: ioctl/client.go [hunk 4]
```diff
@@ -72,6 +76,8 @@ type (
 		conn *grpc.ClientConn
 		// TODO: merge into config
 		lang config.Language
+		// cryptoSm2 is a flag for sm2 cryptographic algorithm
```
Comment (Liuhaai) [inline ioctl/client.go:79]: remove the comment here

File: ioctl/client.go [hunk 5]
```diff
@@ -59,6 +59,10 @@ type (
 		PrintError(error)
 		// PrintInfo print the command result or the question query
 		PrintInfo(string)
+		// SetCryptoSm2 set the flag of sm2 cryptographic algorithm
+		SetCryptoSm2(bool)
+		// GetCryptoSm2 get the flag of sm2 cryptographic algorithm
+		GetCryptoSm2() bool
```
Comment (Liuhaai) [inline ioctl/client.go:65]: `HasCryptoSm2`

**Review by Liuhaai — APPROVED** [review]

File: ioctl/client.go (hunk 4 above)
Comment (huof6829) [reply ioctl/client.go]: ok

File: ioctl/client.go (hunk 5 above)
Comment (huof6829) [reply ioctl/client.go]: yes

**Review by CoderZhi — APPROVED** [review]

File: ioctl/newcmd/account/accountdelete.go (hunk 2 above)
Comment (huof6829) [reply ioctl/newcmd/account/accountdelete.go]: > is this a constant? if not, how could you prevent it be modified in other packages?

this is outdated code. Now move CryptoSm2 into ioctl.Client,  use  client.HasCryptoSm2()  to get status ,  use client.SetCryptoSm2(bool)  in main function

Comment (huof6829) [reply ioctl/newcmd/account/accountdelete.go]: > is this a constant? if not, how could you prevent it be modified in other packages?

 this is outdated code. Now move CryptoSm2 into ioctl.Client, 
 user test the result by setting corresponding gomock function.  HasCryptoSm2() to get status , SetCryptoSm2(bool) to set status

**Pull request merged**